### DISCUSSION TITLE
Pass specialized replication scheme to driver

### DISCUSSION THREAD

**Discussion by bgorbuntsov** [awslabs/aws-mysql-jdbc#181] [Q&A] [answered]
Hi,
I'm stuck with enabling distributing read traffic to replicas while using the AWS JDBC Driver.
The driver documentation (readme.md here) says "Update the connection URL prefix from jdbc:mysql:// to jdbc:mysql:aws://.".
But for enabling replication scheme I need to put "jdbc:mysql:replication://" as prefix for connection URL.
How can I utilize native MySQL driver read traffic distribution and awesome failover functionality of the AWS driver simultaneously?

**Comment by sergiyvamz** [comment, marked as answer]
Hello @bgorbuntsov,

Unfortunately, the AWS JDBC Driver for MySQL only supports failover functionality for the "jdbc:mysql:aws://" protocol. During a cluster failover, the driver picks up a new elected writer node or a random reader node, if eligible, and keeps that connection. The driver doesn't 
switch the connection to distribute a read load. The driver, however, supports the original "jdbc:mysql:replication://" protocol as it's been inherited from the community MySQL JDBC driver, but with no advanced failover support. 

Read load balancing is on our list for future releases; however, we cannot provide exact timelines at the moment. We'll reach out when we have more information.

Thanks!
